util/httputil/http_util.go

```diff
@@ -24,17 +25,14 @@ func HttpGet(url string, auth *config.Auth, timeout time.Duration) ([]byte, int,
 	if err != nil {
 		return nil, 0, err
 	}
-	var client http.Client
-	if auth != nil {
-		transport, err := AuthTransport(auth, &http.Transport{})
-		if err != nil {
-			return nil, 0, err
-		}
-		client = http.Client{Transport: transport, Timeout: timeout}
-	} else {
-		client = http.Client{Timeout: timeout}
+
+	transport, err := CreateTransport(auth, &http.Transport{}, timeout)
```

Comment:
IIUC the one big difference we have with the Prometheus client, compared to the other client calls we do, is that here the same client is reused for all connections.
So, still referring to this https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/ , what was really missing here I guess was the `http.Transport.IdleConnTimeout` which is what impacts the time spent between two requests when the connection pool is full (i guess). All the rest was already covered by the top-level client timeout.

Anyway, even if it covers more than necessary, +1 to set all granular timeouts as a security (if it hasn't any real drawback, but I'm not aware of any)